src/main/java/com/google/api/codegen/discovery/config/ApiaryConfigToSampleConfigConverter.java

```diff
@@ -141,18 +143,23 @@ private MethodInfo createMethod(Method method) {
    */
   private FieldInfo createFieldInfo(Field field, Type containerType, Method method) {
     String placeholder = "";
+    boolean isPlaceholderSingular = true;
     TypeInfo typeInfo = createTypeInfo(field, method);
     if (typeInfo.kind() == Field.Kind.TYPE_STRING && !typeInfo.isArray() && !typeInfo.isMap()) {
       String pattern = apiaryConfig.getFieldPattern().get(containerType.getName(), field.getName());
       if (!Strings.isNullOrEmpty(pattern)) {
-        placeholder = DefaultString.getPlaceholder(field.getName(), pattern);
+        placeholder = DefaultString.getNonTrivialPlaceholder(pattern);
+        // String placeholders are always contained within braces, so counting
+        // the number of open braces is an easy test for singularity.
+        isPlaceholderSingular = StringUtils.countMatches(placeholder, '{') < 2;
```

Comment:
Could there be  a case where the count could be zero, and we print a TODO when there's nothing to replace? Or would this never happen because there wouldn't be a pattern in that case?